util/kube/ctl.go

```diff
@@ -84,6 +91,11 @@ func (k KubectlCmd) ApplyResource(config *rest.Config, obj *unstructured.Unstruc
 	return strings.TrimSpace(string(out)), nil
 }
 
+func needAuthReconcile(obj *unstructured.Unstructured) bool {
```

Comment:
Don't need this function if we always auth reconcile and then apply.

Reply:
Oops, based on Alex's comments we do need this function to check the api version.